Index: packages/openwrt/net/iputils/Makefile
===================================================================
--- packages/openwrt/net/iputils/Makefile	(revision 28942)
+++ packages/openwrt/net/iputils/Makefile	(working copy)
@@ -15,7 +15,7 @@
 PKG_SOURCE_URL:=http://www.skbuff.net/iputils
 PKG_MD5SUM:=a36c25e9ec17e48be514dc0485e7376c
 
-PKG_BUILD_DEPENDS:=sysfsutils
+PKG_BUILD_DEPENDS:=sysfsutils openssl
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-s$(PKG_VERSION)
 
 include $(INCLUDE_DIR)/package.mk
