diff -ruN openvswitch-1.3.0.orig//ovsdb/log.c openvswitch-1.3.0/ovsdb/log.c
--- openvswitch-1.3.0.orig//ovsdb/log.c	2012-01-30 16:59:14.004014376 +0100
+++ openvswitch-1.3.0/ovsdb/log.c	2012-01-30 16:59:29.216014092 +0100
@@ -29,7 +29,7 @@
 #include "lockfile.h"
 #include "ovsdb.h"
 #include "ovsdb-error.h"
-#include "sha1.h"
+#include "lib/sha1.h"
 #include "socket-util.h"
 #include "transaction.h"
 #include "util.h"
diff -ruN openvswitch-1.3.0.orig//tests/test-sha1.c openvswitch-1.3.0/tests/test-sha1.c
--- openvswitch-1.3.0.orig//tests/test-sha1.c	2012-01-30 16:59:14.020014374 +0100
+++ openvswitch-1.3.0/tests/test-sha1.c	2012-01-30 17:48:43.447958732 +0100
@@ -15,7 +15,7 @@
  */
 
 #include <config.h>
-#include "sha1.h"
+#include "lib/sha1.h"
 #include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ruN openvswitch-1.3.0.orig//vswitchd/bridge.c openvswitch-1.3.0/vswitchd/bridge.c
--- openvswitch-1.3.0.orig//vswitchd/bridge.c	2012-01-30 16:59:14.016014374 +0100
+++ openvswitch-1.3.0/vswitchd/bridge.c	2012-01-30 17:53:57.571952847 +0100
@@ -37,7 +37,7 @@
 #include "ofpbuf.h"
 #include "ofproto/ofproto.h"
 #include "poll-loop.h"
-#include "sha1.h"
+#include "lib/sha1.h"
 #include "shash.h"
 #include "socket-util.h"
 #include "stream-ssl.h"
