--- a/src/lxc/lxc-checkconfig.in
+++ b/src/lxc/lxc-checkconfig.in
@@ -27,14 +27,24 @@ is_set() {
     return $?
 }
 
+is_static() {
+    $GREP -q "$1=[y]" $CONFIG
+    return $?
+}
+
 is_enabled() {
     mandatory=$2
 
     is_set $1
     RES=$?
 
-    if [ $RES -eq 0 ]; then
-	$SETCOLOR_SUCCESS && echo -e "enabled" && $SETCOLOR_NORMAL
+    is_static $1
+    STATIC=$?
+
+    if [ $STATIC -eq 0 ]; then
+        $SETCOLOR_SUCCESS && echo -e "static" && $SETCOLOR_NORMAL
+    elif [ $RES -eq 0 ]; then
+	$SETCOLOR_SUCCESS && echo -e "modular" && $SETCOLOR_NORMAL
     else
 	if [ ! -z "$mandatory" -a "$mandatory" = yes ]; then
 	    $SETCOLOR_FAILURE && echo -e "required" && $SETCOLOR_NORMAL
