Index: lxc-r2012092301/src/lxc/conf.c
===================================================================
--- lxc-r2012092301.orig/src/lxc/conf.c	2012-09-24 12:13:41.010913419 +0200
+++ lxc-r2012092301/src/lxc/conf.c	2012-09-24 12:18:30.407164101 +0200
@@ -1865,13 +1865,14 @@
 {
 	char peer[IFNAMSIZ];
 	int err;
+	static uint16_t vlan_cntr = 0;
 
 	if (!netdev->link) {
 		ERROR("no link specified for vlan netdev");
 		return -1;
 	}
 
-	err = snprintf(peer, sizeof(peer), "vlan%d", netdev->priv.vlan_attr.vid);
+	err = snprintf(peer, sizeof(peer), "vlan%d-%d", netdev->priv.vlan_attr.vid, vlan_cntr++);
 	if (err >= sizeof(peer)) {
 		ERROR("peer name too long");
 		return -1;
