Index: lxc-0.7.5/config/linux.m4
===================================================================
--- lxc-0.7.5.orig/config/linux.m4	2012-09-22 21:51:01.302915078 +0200
+++ lxc-0.7.5/config/linux.m4	2012-09-22 21:51:21.335038251 +0200
@@ -66,6 +66,7 @@
 	powerpc*-*) LINUX_SRCARCH=powerpc;;
 	s390*-*) LINUX_SRCARCH=s390;;
 	arm*-*) LINUX_SRCARCH=arm;;
+	mips*-*) LINUX_SRCARCH=mips;;
 	*) AC_MSG_ERROR([architecture ${host} not supported]);;
 	esac
 
